lib/buffer.js

```diff
@@ -392,17 +392,25 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
-Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+Buffer.prototype.indexOf = function indexOf(val, byteOffset, encoding) {
   if (byteOffset > 0x7fffffff)
     byteOffset = 0x7fffffff;
   else if (byteOffset < -0x80000000)
     byteOffset = -0x80000000;
   byteOffset >>= 0;
 
-  if (typeof val === 'string')
-    return binding.indexOfString(this, val, byteOffset);
+  if (typeof val === 'string') {
+    if (encoding === undefined ||
+        encoding === 'ucs2' ||
+        encoding === 'utf16le' ||
+        encoding === 'utf8') {
```

Comment:
the encoding is case insensitive. look at `Buffer.isEncoding()` to see how we do it now to speed up the common case.